### PR TITLE
Update workspace delete command docs to reference RUM vs empty state

### PR DESCRIPTION
**Pull request by JarrettSpiker**
In terraform cloud we are adding some guardrails to help prevent users from accidentally deleting workspaces with resources under management (RUM).

The Terraform CLI already attempts to prevent the user from deleting workspaces with RUM, and requires the `-force` flag to remove a workspace with RUM. The docs + help text around this refer to workspaces which are dangerous to delete as "workspaces with non-empty state".

In order to bring the CLI experience up to date with the TFC/E experience, and standardize on an industry term rather than something terraform specific, this updates the docs/help to refer to "resources under management" instead.

Proposed doc change [here](https://github.com/hashicorp/terraform-docs-common/pull/146) and some changes to the CLI to utilize the Terraform Cloud enhancements [here](https://github.com/hashicorp/terraform/pull/31949)

### CODE REVIEW

File: internal/command/workspace_delete.go
Comment (laurapacilio) [inline internal/command/workspace_delete.go:213]:
```suggestion
  -force             Remove a workspace even if it is managing resources. Terraform can no longer track or manage the workspace's infrastructure. 
```

File: website/docs/cli/commands/workspace/delete.mdx
Comment (laurapacilio) [inline website/docs/cli/commands/workspace/delete.mdx:34]:
```suggestion
* `-force` - Delete the workspace even if it has resources under management. After deletion, Terraform can no longer track or manage the workspace's infrastructure. Defaults to false.
```

**Review by laurapacilio — APPROVED** [review]
Approved with a few nits. i think we should try to help users understand the implications of this action (and why it might be dangerous to do it).

I would also like someone from the core team to approve this before you merge it :) Possibly @apparentlymart - just because it touches .go files. 

Finally, I'm going to add a backport label to this so it'll show up in latest after you merge. You'll need to also merge the auto-generated backport PR. Let me know if you have questions about that!

**Comment by JarrettSpiker** [comment]
I am removing the 1.3-backport flag, since the corresponding feature is only going into 1.4

@brandonc would you mind taking a look at this small CLI helptext change which goes along with the recent changes to `workspace delete -force` when you get a chance?

File: website/docs/cli/commands/workspace/delete.mdx
Comment (brandonc) [inline website/docs/cli/commands/workspace/delete.mdx:17]: It turns out that even without considering Terraform Cloud workspaces, this is more correct because the requirement is that the state be empty of any resources (not outputs or metadata)

However, Terraform CLI seems to prefer the word "tracked" to "managed" today. I wonder if we should mirror that term here?

```suggestion
To delete a workspace, it must already exist, it must not be tracking resources,
and it must not be your current workspace. If the workspace is tracking resources,
```

Screen Shot 2022-11-22 at 12 47 16 PM

File: website/docs/cli/commands/workspace/delete.mdx
Comment (brandonc) [inline website/docs/cli/commands/workspace/delete.mdx:20]: Should we introduce this acronym? Is it a term we want to familiar users with to the extend that we want to use it in the docs? It seems like we usually spell out words like Terraform Cloud instead of using acronyms like TFC

File: website/docs/cli/commands/workspace/delete.mdx
Comment (laurapacilio) [reply website/docs/cli/commands/workspace/delete.mdx:17]: ah, this is an older draft! @JarrettSpiker product management let me know we're not saying RUM anymore, but more just "managed resources".  So I support removing!

File: website/docs/cli/commands/workspace/delete.mdx
Comment (JarrettSpiker) [reply website/docs/cli/commands/workspace/delete.mdx]: good feedback! For a while there it seemed like we were going to make RUM a thing, but since we are not I have updated this to talk about tracked resources instead, as per your other comment

**Review by brandonc — APPROVED** [review]

File: internal/command/workspace_delete.go
Comment (alisdair) [inline internal/command/workspace_delete.go:213]: We need to wrap this help text manually, as Terraform doesn't perform word wrapping for us. Without word wrap it breaks awkwardly mid-word at the terminal edge:

image

I think we manually wrap at 76 characters by convention.

**Review by alisdair — CHANGES_REQUESTED** [review]

**Review by alisdair — APPROVED** [review]

**Pull request merged**
